app/src/main/java/com/github/drawyourpath/bootcamp/path/Run.kt

```diff
@@ -0,0 +1,88 @@
+package com.github.drawyourpath.bootcamp.path
+
+import com.firebase.ui.auth.data.model.User
+
+
+/**
+ * A class representing a run taken by a user.
+ * The run includes information such as the path taken, distance, duration, timestamp,
+ * user, averageSpeed, and calorie burn.
+ * Various methods are included to calculate and retrieve this information.
+ */
+class Run (
+    //val user: User,             //TODO add later
+    private val path: Path,       //represents the path taken by the user
+    private val startTime: Long,  //the timestamps of the run
+    private val endTime: Long
+        ){
+
+
+    init {
+        if (endTime <= startTime) {
+            throw IllegalArgumentException("End time must be greater than start time")
+        }
+        calculateDistance()
+        calculateDuration()
+        calculateAverageSpeed()
+    }
+
+
+
+    //the distance of the run (in meters)
+    private var distance: Double = 0.0
+    //the duration of the run (in seconds)
+    private var duration: Long = 0L
+    //the average speed of the run (in meters per second)
+    private var averageSpeed: Double = 0.0
+    //the calorie burn of the run
+    private var calories: Int = 0
+

```

Comment:
(Maybe you can add the time made by the runner in (minutes, seconds) to make 1 kilometer, this measure is used in the most of the running apps)